### PR TITLE
Add flag validation for WGE commands

### PR DESCRIPTION
**Pull request by yiannistri**
Closes: weaveworks/weave-gitops-enterprise#207

**What changed?**
- Refactor root command to allow for easier unit testing
- Set `SilenceUsage`/`SilenceErrors` for all WGE commands to `true`
- Added validation for endpoint and provider

**Why?**
To improve errors returned to the user

**How did you test it?**
Local testing + new unit tests

**Release notes**
N/A

**Documentation Changes**
N/A

### CODE REVIEW

File: pkg/wegoerrors/errors.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package wegoerrors
+
+import "errors"
+
+var ErrWGEHTTPApiEndpointNotSet = errors.New("the Weave GitOps Enterprise HTTP API endpoint flag (--endpoint) has not been set")
```
Comment (bigkevmcd) [inline pkg/wegoerrors/errors.go:5]: I suspect you could shorten this to just `ErrWGEEndpointNotSet` and maybe even `ErrNoWGEEndpoint` which would be more in keeping with the Go stdlib https://pkg.go.dev/net/http#pkg-variables and https://pkg.go.dev/io#pkg-variables

Comment (bigkevmcd) [reply pkg/wegoerrors/errors.go:5]: Oh, and I think this should be in the `cmd` package ?

Because it explicitly refers to a command-line option `--endpoint)` ?

**Comment by foot** [comment]
Does this change any behaviour in the commands like `gitops add app` etc? 🤔

**Comment by yiannistri** [comment]
I don't think it does as only the WGE commands were updated 🤔

File: pkg/wegoerrors/errors.go (hunk 1 above)
Comment (yiannistri) [reply pkg/wegoerrors/errors.go]: Renamed and moved to `cmd` package.

File: cmd/gitops/cmderrors/errors.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package cmderrors
```
Comment (jpellizzari) [inline cmd/gitops/cmderrors/errors.go:1]: :+1:

**Review by jpellizzari — APPROVED** [review]

**Review by foot — APPROVED** [review]
💯
